### DISCUSSION TITLE
Feature-Request: allow vcpkg initial-installation via Chocolatey (package manager)

### DISCUSSION THREAD

**Discussion by DavidLudwig** [microsoft/vcpkg#17785] [Ideas]
The vcpkg bootstrap scripts on Windows seems reasonably nice, however, it might be a bit more convenient to install via the [Chocolatey package manager](https://chocolatey.org/), using a (potential) command such as the following:

```
choco install vcpkg
```

**Comment by degski** [comment]
@DavidLudwig 

You are right.

    choco install vcpkg

is a lot nicer than:

    bootstrap-vcpkg

A question, though, which package-manager do I use to install the package-manager `chocolatey` to install the package manager `vcpkg`?

**Comment by DavidLudwig** [comment]
Chocolatey's install process is done via it's own installer script.  I find that its package catalog makes it a worthwhile install, in and of itself (with lots of Microsoft things in it!)

What about installation of vcpkg to one's PATH, via the bootstrap accept?  I might be able to write something up for that, say, in GitHub Issues, if that'd be preferable.  I have a hard time seeing some developers, especially beginners, dealing with the need to figure out PATH, or whatever its equivalent is in PowerShell (I am more of a command.com old-timer).

**Comment by degski** [comment]
@DavidLudwig 

> What about installation of vcpkg to one's PATH, via the bootstrap accept?

The path is not hard, though, if one cannot learn even the most basic OS stuff, there's no use in trying to tackle C++.

    set PATH=%PATH%;C:\some_path;

There is [`Varpanel`](http://implbits.com/products/varpanel/), very useful and usable by beginners. There is also [`HashTab`](http://implbits.com/products/hashtab/), from the same company, equally useful.

> ... or whatever its equivalent is in PowerShell ...

There's no need to use PS (all works equally fine from a developer-cmd-prompt). I'm also more a cmd man, but, [`Windows Powershell ISE`](https://docs.microsoft.com/en-us/powershell/scripting/core-powershell/ise/introducing-the-windows-powershell-ise?view=powershell-6) is freakin' nice.

I'll check out `chocolatey`.

**Comment by DavidLudwig** [comment]
Cool, and thanks for the tips!

**Comment by MVoz** [comment]
the best edit Environment 
https://www.rapidee.com/en/screenshots

**Comment by pr8x** [comment]
Any update on this issue?

**Comment by Foadsf** [comment]
@degski Let's put the question this way, can vcpkg update itself? if yes, you probably don't need chocolatey that much. Otherwise, it would really be helpfull. Plus installing via choco or homebrew is easier than cloning the git repo and compiling IMHO.

**Comment by isubasinghe** [comment]
@Foadsf its better to have a central place to download windows packages still imo.

**Comment by Honeybunch** [comment]
✋ I'd really like vcpkg in chocolatey and other package managers.
Rust, Dlang and other languages have a similar system. You install cargo or dub from your package manager and use those to interact with packages for the language ecosystem you're working in. I think it makes a lot of sense.

**Comment by tyteen4a03** [comment]
Any updates on this?

**Comment by ThomasEricB** [comment]
We need this ASAP

**Comment by autoantwort** [comment]
Can chocolatey install a complete git repository? And does an update delete the installed folder?
